controllers/discovery/searchUser.js

```diff
@@ -23,6 +23,14 @@ const SearchUser = async (req, res) => {
     where_anon_dm = 'AND u.allow_anon_dm = :allow_anon_dm';
   }
 
+  const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(UserBlockedUser, req.userId);
+  let filterBlockedUser = '';
+  if (blockedIds.length > 0) {
+    filterBlockedUser += `AND u.user_id NOT IN (${blockedIds
+      .map((item) => `'${item}'`)
+      .join(',')})`;
+  }
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code retrieves the IDs of blocked users and constructs a SQL clause to exclude these users from the search results. This is a good approach to ensure that blocked users are not included in the search results. However, it's important to note that this could potentially lead to SQL injection if the `blockedIds` are not properly sanitized or validated. Make sure that the `getBlockedAndBlockerUserId` function returns only valid user IDs.
```diff
+ const blockedIds = await UsersFunction.getBlockedAndBlockerUserId(UserBlockedUser, req.userId);
+ let filterBlockedUser = '';
+ if (blockedIds.length > 0) {
+   filterBlockedUser += `AND u.user_id NOT IN (${blockedIds
+     .map((item) => `'${item}'`)
+     .join(',')})`;
+ }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->